src/script_opt/CPP/RuntimeVec.cc

```diff
@@ -103,20 +104,25 @@ VEC_OP1(comp, ~, )
 
 // A kernel for applying a binary operation element-by-element to two
 // vectors of a given low-level type.
-#define VEC_OP2_KERNEL(accessor, type, op)                                                         \
+#define VEC_OP2_KERNEL(accessor, type, op, zero_check)                                             \
```

Comment:
I know this isn't related to this PR, but I'd really love to see if we can turn these into lambdas at some point instead of macros.

Reply:
Ah yeah maybe we can.  I'm not used to thinking in terms of lambdas for C++ stuff so went the primitive route here, but point taken.